nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/media/ExtractMediaMetadata.java

```diff
@@ -0,0 +1,311 @@
+        + "extracts a long list of metadata types for media files including audio, video, and print media "
+        + "formats."
+        + "For the more details and the list of supported file types, visit the library's website "
+        + "at http://tika.apache.org/.")
+@WritesAttributes({@WritesAttribute(attribute = "<Metadata Key Prefix>.<attribute>", description = "The extracted content metadata "
+        + "will be inserted with the attribute name \"<Metadata Key Prefix>.<attribute>\", or \"<attribute>\" if "
+        + "\"Metadata Key Prefix\" is not provided.")})
+@SupportsBatching
+public class ExtractMediaMetadata extends AbstractProcessor {
+
+    static final PropertyDescriptor MAX_NUMBER_OF_ATTRIBUTES = new PropertyDescriptor.Builder()
+            .name("Max Number of Attributes")
+            .description("Specify the max number of attributes to add to the flowfile. There is no guarantee in what order"
+                    + " the tags will be processed. By default it will process all of them.")
+            .required(false)
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+    private static final PropertyDescriptor MAX_ATTRIBUTE_LENGTH = new PropertyDescriptor.Builder()
+            .name("Max Attribute Length")
+            .description("Specifies the maximum length of a single attribute value.  When a metadata item has multiple"
+                    + " values, they will be merged until this length is reached and then \", ...\" will be added as"
+                    + " an indicator that additional values where dropped.  If a single value is longer than this, it"
+                    + " will be truncated and \"(truncated)\" appended to indicate that truncation occurred.")
+            .required(true)
+            .defaultValue("10240")
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor MIME_TYPE_FILTER = new PropertyDescriptor.Builder()
+            .name("MIME Type Filter")
+            .description("A regular expression identifying MIME types for which metadata should extracted.  Flowfiles"
+                    + " selected for scanning by the File Name Filter are parsed to determine the MIME type and extract"
+                    + " metadata.  If the MIME type found matches this regular expression or this expression is"
+                    + " blank, the metadata keys that match the Metadata Key Filter will be added to the flowfile"
+                    + " attributes.  There is no guarantee in what order attributes will be produced.  If"
+                    + " left blank, metadata will be extracted from all flow files selected for scanning.")
+            .required(false)
+            .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
+            .build();
```

Comment:
Then why wouldn't they use IdentifyMimeType (which uses a Tika Detector) to identify the mime type and route it away if needed? Given that the detector is just detecting the content I believe it would be much faster than parsing the entire thing.